httprouter/apirest/apirest.go

```diff
@@ -142,6 +169,15 @@ func (a *API) RegisterMethod(pattern, HTTPmethod string,
 	routerHandler := func(msg httprouter.Message) {
 		bsaMsg := msg.Data.(*APIdata)
 		if err := handler(bsaMsg, msg.Context); err != nil {
+			// if err is an APIError, use those properties
+			if apierror, ok := err.(APIerror); ok {
+				err := msg.Context.Send(apierror.JSON(), apierror.HTTPstatus)
```

Comment:
I'd reverse the functionallity.

The APIerror type might have a method named `Send` that takes the `msg.Context` as a parameter.
Then you can call `apierror.Send(msg.Context)` directly, which is nicer.